deepmd/utils/data.py

```diff
@@ -440,6 +468,30 @@ def _shuffle_data(self, data):
                 ret[kk] = data[kk]
         return ret, idx
 
+    def _get_nframes(self, set_name: DPPath):
+        # get nframes
+        if not isinstance(set_name, DPPath):
+            set_name = DPPath(set_name)
+        path = set_name / "coord.npy"
+        if self.data_dict["coord"]["high_prec"]:
+            coord = path.load_numpy().astype(GLOBAL_ENER_FLOAT_PRECISION)
+        else:
+            coord = path.load_numpy().astype(GLOBAL_NP_FLOAT_PRECISION)
+        if coord.ndim == 1:
+            coord = coord.reshape([1, -1])
+        nframes = coord.shape[0]
+        return nframes
+
+    def preprocess(self, data):
```

Comment:
doc string needed. 
is the preprocess used only for the torch code? 
suggest to rename the function in a more self-explanatory way.